homeassistant/components/marstek/scanner.py

```diff
@@ -0,0 +1,200 @@
+            for device in devices:
+                _LOGGER.debug(
+                    "  Device: %s at IP %s (BLE-MAC: %s)",
+                    device.get("device_type", "Unknown"),
+                    device.get("ip", "Unknown"),
+                    device.get("ble_mac", "N/A"),
+                )
+
+            # Check all configured entries for IP changes
+            # Check both LOADED and SETUP_RETRY states (SETUP_RETRY means connection failed)
+            for entry in self._hass.config_entries.async_entries(DOMAIN):
+                _LOGGER.debug(
+                    "Scanner: Checking entry %s (state: %s)",
+                    entry.title,
+                    entry.state,
+                )
+                if entry.state not in (
+                    ConfigEntryState.LOADED,
+                    ConfigEntryState.SETUP_RETRY,
+                ):
+                    _LOGGER.debug(
+                        "Scanner: Skipping entry %s - state is %s (not LOADED)",
+                        entry.title,
+                        entry.state,
+                    )
+                    continue
+
+                stored_ble_mac = entry.data.get("ble_mac")
+                stored_ip = entry.data.get(CONF_HOST)
+
+                _LOGGER.debug(
+                    "Scanner: Entry %s - stored BLE-MAC: %s, stored IP: %s",
+                    entry.title,
+                    stored_ble_mac or "N/A",
+                    stored_ip or "N/A",
+                )
+
+                if not stored_ble_mac or not stored_ip:
+                    _LOGGER.debug(
+                        "Scanner: Skipping entry %s - missing BLE-MAC or IP",
+                        entry.title,
+                    )
+                    continue
+
+                # Find matching device by BLE-MAC
+                matched_device = self._find_device_by_ble_mac(
+                    devices, stored_ble_mac, entry.title
+                )
+
+                if not matched_device:
+                    _LOGGER.debug(
+                        "Scanner: No matching device found for entry %s (BLE-MAC: %s)",
+                        entry.title,
+                        stored_ble_mac,
+                    )
+                    continue
+
+                new_ip = matched_device.get("ip")
+                _LOGGER.debug(
+                    "Scanner: Entry %s - current IP: %s, discovered IP: %s",
+                    entry.title,
+                    stored_ip,
+                    new_ip,
+                )
+                if new_ip and new_ip != stored_ip:
+                    _LOGGER.info(
+                        "Scanner detected IP change for device %s: %s -> %s",
+                        stored_ble_mac,
+                        stored_ip,
+                        new_ip,
+                    )
+                    # Update config entry via config flow (mik-laj feedback)
+                    self._hass.config_entries.async_update_entry(
```

Comment:
We should trigger discovery flow. Here is related code in Yeelight.
https://github.com/home-assistant/core/blob/319d6711c4fa4c6f93c4149d2824ad273f6469d8/homeassistant/components/yeelight/scanner.py#L171-L181
And this flow should update config entries.
https://github.com/home-assistant/core/blob/319d6711c4fa4c6f93c4149d2824ad273f6469d8/homeassistant/components/yeelight/config_flow.py#L96-L123